plugins/python-build/scripts/add_miniconda.py

```diff
@@ -140,12 +147,23 @@ def from_str(cls, s):
         return MinicondaVersion(Suffix(miniconda_n[-1]), VersionStr(ver), py_ver)
 
     def to_filename(self):
+        """
+        Convert a miniconda version string to a filename.
+        """
         if self.py_version:
             return f"miniconda{self.suffix}-{self.py_version.version()}-{self.version_str}"
         else:
             return f"miniconda{self.suffix}-{self.version_str}"
 
     def default_py_version(self):
+        """
+        Returns the Python version to use for a given conda environment.
+
+        If `py_ver` is provided, it will be used as the Python version. Otherwise, if
+        `suffix` is provided and ends with '27', then Python 2.7 will be used; otherwise, if `conda_info` indicates that Conda 4.7 or later is installed
+        (i.e., its info() method returns an int >= 4), then Python 3.6+ will be returned; otherwise, 3<=Python<3.<8> will be returned (2 and <3 are both
+        interpreted as 2.*).
```

Comment:
No point duplicating the code literally line by line (and the explanation is more confusing than the code, too).

Reply:
Removed long explanation.